modules/irsa/main.tf

```diff
@@ -1,5 +1,12 @@
 locals {
   eks_oidc_issuer_url = replace(var.eks_oidc_provider_arn, "/^(.*provider/)/", "")
+
+  iam_role_name = try(coalesce(var.irsa_iam_role_name, format("%s-%s-%s", var.eks_cluster_id, trim(var.kubernetes_service_account, "-*"), "irsa")), null)
+
+  # This ensures that the IAM role does not exceed 64 characters which is the current limitation of IAM roles
+  # https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html
+  # tflint-ignore: terraform_unused_declarations
+  iam_role_name_length_check = range(0, 65)[length(local.iam_role_name)]
```

Comment:
isn't it better to use `substr`?

Reply:
@FernandoMiguel I'm not sure I'm following. Could you provide an example?

Reply:
that's why i almost always use `name_prefix` and truncate be provided user name

Reply:
```
resource "aws_iam_role" "awslbc_irsa" {
  name_prefix        = "${substr(var.eks_cluster_name, 0, 25)}-awslbc-irsa-"
  description        = "IRSA role for AWS Load Balancer Controller for EKS cluster ${var.eks_cluster_name}"
```

Reply:
and yeah, been bitten by the trailing dash once.... ugh

Reply:
Unfortunately `name_prefix` would cause a ForceNew here

Reply:
The alternative as @lorengordon pointed out is to use a `precondition` hook

```hcl
  lifecycle {
    # This ensures that the IAM role does not exceed 64 characters which is the current limitation of IAM roles
    # https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html
    precondition {
      condition     = length(local.iam_role_name) <= 64
      error_message = "The IAM Role Name cannot be greater than 64 characters. iam_role_name = ${local.iam_role_name}."
    }
  }
```

This would work well but unfortunately it requires a minimum terraform version of 1.2 and this module uses a minimum terraform version of 1.0.

https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/16ee228f857deb00cb1015085588694f00a1ecd4/versions.tf#L2

Reply:
@FernandoMiguel could I get another review?

Reply:
not a project maintainer. 
i really dont like this approach tho